packages/xchain-bitcoin/src/client.ts

```diff
@@ -249,88 +252,101 @@ class Client implements BitcoinClient, XChainClient {
-  //   const balance = this.getBalance()
-  //   const balancePlaceholder = balance - Utils.dustThreshold - 1
-  //   const psbt = new Bitcoin.Psbt({ network: network }) // Network-specific
-  //   this.utxos.forEach((UTXO) =>
-  //     psbt.addInput({
-  //       hash: UTXO.hash,
-  //       index: UTXO.index,
-  //       witnessUtxo: UTXO.witnessUtxo,
-  //     }),
-  //   )
-  //   psbt.addOutput({ address: addressTo, value: balancePlaceholder }) // Add output
-  //   psbt.addOutput({ address: this.getAddress(), value: 1 }) // change output
-  //   if (memo) {
-  //     const data = Buffer.from(memo, 'utf8') // converts MEMO to buffer
-  //     const OP_RETURN = Bitcoin.script.compile([Bitcoin.opcodes.OP_RETURN, data]) // Compile OP_RETURN script
-  //     psbt.addOutput({ script: OP_RETURN, value: 0 }) // Add OP_RETURN {script, value}
-  //   }
-  //   psbt.signAllInputs(btcKeys) // Sign all inputs
-  //   const tx = psbt.finalizeAllInputs().extractTransaction() // Finalise inputs, extract tx
-  //   const inputs = this.utxos.length // Add weight for each input sig
-  //   return tx.virtualSize() + inputs
-  // }
+  /**
+   * Calculates fees based on fee rate and memo
+   */
+  calcFee = (feeRate: FeeRate, memo?: string): BaseAmount => {
+    if (memo) {
+      const OP_RETURN = Utils.compileMemo(memo)
+      const vaultFee = Utils.getVaultFee(this.utxos, OP_RETURN, feeRate)
+      return baseAmount(vaultFee)
+    }
+    const normalFee = Utils.getNormalFee(this.utxos, feeRate)
+    return baseAmount(normalFee)
+  }
 
   /**
-   * getFees
+   * Returns rates and fees
    */
-  async getFees(): Promise<Fees> {
+  getFeeData = async (memo?: string): Promise<FeeData> => {
     await this.scanUTXOs()
     if (this.utxos.length === 0) {
       throw new Error('No utxos to send')
     }
 
     const btcStats = await blockChair.bitcoinStats(this.nodeUrl, this.nodeApiKey)
     const nextBlockFeeRate = btcStats.suggested_transaction_fee_per_byte_sat
-    const feesOptions: Fees = {
+    const rates: Record<FeeOptionKey, number> = {
+      fastest: nextBlockFeeRate * 5,
+      fast: nextBlockFeeRate * 1,
+      average: nextBlockFeeRate * 0.5,
+    }
+
+    const fees: Fees = {
       type: 'byte',
-      fastest: baseAmount(5),
-      average: baseAmount(1),
-      fast: baseAmount(0.5),
+      fast: this.calcFee(rates.fast, memo),
+      average: this.calcFee(rates.average, memo),
+      fastest: this.calcFee(rates.fastest, memo),
     }
-    const calcdFees: Fees = {
-      type: feesOptions.type,
-      fast: baseAmount(feesOptions.fast?.amount().multipliedBy(nextBlockFeeRate)),
-      average: baseAmount(feesOptions.average?.amount().multipliedBy(nextBlockFeeRate)),
-      fastest: baseAmount(feesOptions.fastest?.amount().multipliedBy(nextBlockFeeRate)),
+
+    return { fees, rates }
+  }
+
+  /**
+   * Returns fees for transactions w/o a memo
+   */
+  getFees = async (): Promise<Fees> => {
```

Comment:
hm... im really not sure that this is makes sense to separate `getFees` and `getFeesWithMemo` as they do the same (and if i understand correctly they WILL do the same all the time) so maybe we can set 
```ts
getFees = async (memo?: string): Promise<Fees>
``` 

the same thought about `getFeeRates/getFeeRatesWithMemo`

Reply:
It's because `interface` definition of `XChainClients` https://github.com/xchainjs/xchainjs-lib/blob/6831adb99d43292bc185d3194f3ad40fddf0e191/packages/xchain-client/src/types.ts#L91

Currently `BitcoinClient` is the only client which needs memos to calculate fees.

Reply:
> the same thought about getFeeRates/getFeeRatesWithMemo

Good point! In this case it does not break any interface of `XChainClients`! However, I would suggest to do it similar to `getFees` / `getFeesWithMemo` , just to be consistent.